src/geometry/manifold/basic_smooth_bundle.lean

```diff
@@ -462,9 +465,20 @@ def tangent_bundle_core : basic_smooth_bundle_core I M E :=
     simp only [A, continuous_linear_map.coe_comp'] with mfld_simps
   end }
 
-/-- The tangent bundle to a smooth manifold, as a plain type. -/
+variable {M}
+
+/-- The tangent space at a point of the manifold `M`. It is just `E`. -/
+def tangent_space (x : M) : Type* :=
+(tangent_bundle_core I M).to_topological_fiber_bundle_core.fiber x
+
+variable (M)
+
+/-- The tangent bundle to a smooth manifold, as a plain type. We could use
+`(tangent_bundle_core I M).to_topological_fiber_bundle_core.total_space`, but instead we use the
+(definitionally equal) `Σ (x : M), tangent_space I x`, to make sure that rcasing an element of the
+tangent bundle gives a second component in the tangent space. -/
 @[nolint has_inhabited_instance] -- is empty if the base manifold is empty
-def tangent_bundle := (tangent_bundle_core I M).to_topological_fiber_bundle_core.total_space
+def tangent_bundle := Σ (x : M), tangent_space I x
```

Comment:
Is it a problem to mark this as `reducible`? This would allow vector bundles to be defined exactly as sigma types

Reply:
Imagine you have two line bundles over the same manifold `M`. Then the total spaces of the two bundles will be `Σ (x : M), ℝ`, but with different topologies and manifold structures. So, in this case, you definitely want to hide these behind type synonyms, and put the topologies and the manifold structure *not* on `Σ (x : M), ℝ`, but only on the type synonyms. For the tangent bundle, the situation is less clear as I don't think we will ever put another manifold structure on `Σ (x : M), tangent_space I x`, but still I don't see what using `reducible` here would buy us.

Reply:
Well, I understand your point, but I had in mind another solution. I understand this is a crazy idea, but my dream is to define vector bundles directly as sigma types, as this would solve all possible fiber problems and everything looks amazing like this (I can PR it so that you can see it as soon as I manage to go on with my PRs). My idea to solve the problem you present is to use the type synonims at the level of fibers (this is not too crazy: it just changes when you use the type synonim), with the following convention: the sigma type of a bare type is meant to have the product topology on it, i.e. is the trivial bundle, for all other cases type synonims must be used. So for example in the case of one dimensional bundles over S^1, `Σ (x : M), ℝ` is the correct name for the cylinder, `Σ (x : M), mobius_fiber` is the correct name for the mobius band. `Σ (x : M), tangent_space x` is the correct name for the tangent bundle and only a posterior you prove they are isomorphic. I understand that the crazy side of this idea is that you are putting a very strong constraint on the implementation on fiber bundles, but I believe there are workarounds when this is not the case (and I can't really imagine cases where this would not be the natural thing to do). In any case the advantages of this implementation are very many, I can try to list them but I believe it is better to see what it would look like in practice. I will just mention that this way it would be very easy to treat in one case riemannian manifolds and riemannian vector bundles, and this way it is very easy to perform operations on vector bundles, such as sum, tensor product, dual, exterior products... what do you think?

Reply:
I definitely think that it is a good idea that all smooth vector bundles have a sigma type as the underlying type. My question is: why is it better to use `Σ (x : M), tangent_space I x` than a type synonym `tangent_space I M` which is more readable for the user? All the constructions you can do with one you can do with the other, as they are the same to Lean kernel.

Reply:
Ok, I believe that for the theory of vector bundles I developed you need instances directly on sigma types, but there is actually no need to discuss this now, let us discuss it when I make my PR, so that maybe you tell me what I am doing wrong so that types synonyms do not work. I will push it asap but first I need to PR diffeos and for that I need to PR smooth bundled maps, that are in que.

Reply:
No problem. I have just checked that adding `reducible` to the definition of the tangent bundle breaks almost nothing, so it will be easy to add when needed.

Reply:
I think I like your idea for vector bundles a lot, so I have just marked the tangent bundle as reducible.

Reply:
Haha ok cool, I will PR it asap and if you will find a way to remove the reducibles I'd be happy to remove them as well btw